### ISSUE TITLE
Example code returns client error

### ISSUE THREAD

**Issue by AugustDev** [argentlabs/web3.swift#157]
I tried to run sample code in readme after importing library using SPM. I connect to mainnet using Infura.

```swift
    guard let clientUrl = URL(string: "https://mainnet.infura.io/v3/MY_TOKEN_HERE) else { return }
    let client = EthereumClient(url: clientUrl)
    
    client.eth_gasPrice(completion: { er, val in
        print(er)
        print(val)
    })
```

`eth_gasPrice` method returns `web3.EthereumClientError.unexpectedReturnValue`

```
Optional(web3.EthereumClientError.unexpectedReturnValue)
nil
```

Any ideas?

**Comment by cboka** [comment]
No, I have no idea!  Does it have anything to do with Metamask?


În joi, 4 nov. 2021 la 08:49 Augustinas Malinauskas <
***@***.***> a scris:

**Comment by rhaining** [comment]
have you tried saving the `client` as a var? it might be getting released

**Comment by UemitGuel** [comment]
Hi @AugustDev, did you find a solution to this problem? 🙂

**Comment by virus-zip** [comment]
Is there any update on this situation? The error doesn't seem to be resolved.

**Comment by virus-zip** [comment]
If there is a print function (yes, a literal print-it-out function) for the client and erc721 instances in the function call, the returned value will be valid. 

print(erc721, client) in the completion block will result in a working getBalance function. I am surprised.

**Comment by DarthMike** [comment]
@AugustDev what is the exact usage of the code? `client` should be owned otherwise it goes out of memory. Did you paste your infura token correctly?

**Comment by johnCbogil** [comment]
This is my exact usage of the code where it fails:
```
guard let clientUrl = URL(string: endpoint) else { return }
let client = EthereumClient(url: clientUrl)
client.eth_getBalance(
            address: account.address,
            block: .Latest) { error, result in
                print(error)
                print(result)
            }
```

However, creating a pointer to the client inside the closure solves the problem. Interesting that an additional pointer is necessary to keep the client alive.

**Comment by DarthMike** [comment]
@johnCbogil The client goes out of memory and a reference should be kept. After the function call `client` will be deallocated, as `eth_getBalance` is asynchronous.

If you use `client` inside the closure, you're effectively keeping it alive until the closure executes (default strong reference). This is normal swift behaviour.
